packages/keyring-controller/src/KeyringController.ts

```diff
@@ -581,26 +611,37 @@ export class KeyringController extends BaseControllerV2<
           `Missing or invalid address ${JSON.stringify(messageParams.from)}`,
         );
       }
-      const qrKeyring = await this.getOrAddQRKeyring();
-      const qrAccounts = await qrKeyring.getAccounts();
-      if (
-        qrAccounts.findIndex(
-          (qrAddress: string) =>
-            qrAddress.toLowerCase() === address.toLowerCase(),
-        ) !== -1
-      ) {
-        const messageParamsClone = { ...messageParams };
+
+      if ((await this.getAccountKeyringType(address)) === KeyringTypes.qr) {
```

Comment:
Starting from `@metamask/eth-keyring-controller@11.0.0` any attempt to add a Keyring of a type with a missing keyring builder throws an error instead of returning undefined.